app/src/main/java/br/com/nespressohelper/daitan/nespressohelper/MainActivity.java

```diff
@@ -311,4 +233,57 @@ public void onClick(View view) {
         });
 
     }
+
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    @Override
+    public void handleResponse(String response, Method m) {
+
+        if(m == Method.GET) {
+            ArrayList<Coffee> coffees = new ArrayList<>();
+            ArrayList<Integer> drawables = generateDrawablesIds();
+
+            coffees = BackEndComm.jsonToObj(response);
+
+            for(int i = 0; i < coffees.size(); i++) {
+                if(i >= drawables.size()) {
+                    drawable = getDrawable(R.drawable.coffecapsule);
+                    bitmap = ((BitmapDrawable) drawable).getBitmap();
+                    stream = new ByteArrayOutputStream();
```

Comment:
I believe I've already saw this exact same code before, you could maybe extract it somewhere